lib/realtime/vehicle.ex

```diff
@@ -53,45 +68,52 @@ defmodule Realtime.Vehicle do
     :direction_id,
     :route_id,
     :trip_id,
+    :bearing,
+    :speed,
+    :stop_sequence,
+    :block_id,
+    :operator_id,
+    :operator_name,
+    :run_id,
     :stop_status,
     :timepoint_status
   ]
 
-  @doc """
-    Argument is an Elixir object. Pass it through Jason before this function.
-    json format for vehicles:
-    {
-      "id": "y0507",
-      "vehicle": {
-        "current_status": "IN_TRANSIT_TO",
-        "current_stop_sequence": 3,
-        "position": {
-          "bearing": 0,
-          "latitude": 42.35277354,
-          "longitude": -71.0593878
-        },
-        "stop_id": "6555",
-        "timestamp": 1554927574,
-        "trip": {
-          "direction_id": 0,
-          "route_id": "505",
-          "schedule_relationship": "SCHEDULED",
-          "start_date": "20190410",
-          "trip_id": "39984755"
+  @spec from_vehicle_position_and_trip_update(map() | nil, map() | nil) :: t()
+  def from_vehicle_position_and_trip_update(nil, _trip_update) do
+    nil
+  end
+
+  def from_vehicle_position_and_trip_update(_vehicle_position, nil) do
+    nil
+  end
+
+  def from_vehicle_position_and_trip_update(
+        %VehiclePosition{
+          id: id,
+          trip_id: trip_id,
+          stop_id: stop_id,
+          label: label,
+          latitude: latitude,
+          longitude: longitude,
+          bearing: bearing,
+          speed: speed,
+          stop_sequence: stop_sequence,
+          block_id: block_id,
+          operator_id: operator_id,
+          operator_name: operator_name,
+          run_id: run_id,
+          last_updated: last_updated,
+          status: status
         },
-        "vehicle": { "id": "y0507", "label": "0507" }
-      }
-    }
-  """
-  @spec decode(term()) :: t()
-  def decode(%{} = json) do
+        %TripUpdate{
+          route_id: route_id,
+          direction_id: direction_id
```

Comment:
Aren't `route_id` and `direction_id` already in `VehiclePosition`, inside the trip descriptor (`trip`)? It seems like it should be unnecessary to have to get trip updates just for the route and direction.

Reply:
No, `VehiclePosition` only has a `trip_id`. Plus, `TripUpdate` isn't something we are having to fetch, it's coming along with the `VehiclePosition` data.

Reply:
It only has a `trip_id` because Concentrate also gets TripUpdates from a predictions feed. For your usecase, you could parse them all into a single struct.

Reply:
Based on what I can tell, keeping the TripUpdate separate from VehiclePosition allows the `route_id`, `direction_id`, and `schedule_relationship` fields to be updated by `StopTimeUpdate`s in addition to being updated by `VehiclePosition`.

That seems like something we don't need.
* We don't consume `StopTimeUpdate`s.
* Wouldn't the fields be updated inside the `VehiclePosition`, too?

A lot of this code would become much simpler if we didn't have separate `TripUpdate`s, so I'd be favor of putting the fields into a single `VehiclePosition` struct.

Reply:
I feel like trying to customize this stuff for our specific app is a bad idea. What we want to do is to move these into a shared library and get them out of our app altogether. Everything we do to change how these work is just going to make that harder.